openai/responses.go

```diff
@@ -0,0 +1,992 @@
+	TopP *float64 `json:"top_p"`
+
+	// optional, default is `"disabled"`
+	Truncation *string `json:"truncation"`
+
+	Tools []ResponsesTool `json:"tools,omitempty"`
+
+	// TODO(drifkin): tool_choice is not supported. We could support "none" by not
+	// passing tools, but the other controls like `"required"` cannot be generally
+	// supported.
+
+	// optional, default is false
+	Stream *bool `json:"stream,omitempty"`
+}
+
+// FromResponsesRequest converts a ResponsesRequest to api.ChatRequest
+func FromResponsesRequest(r ResponsesRequest) (*api.ChatRequest, error) {
+	var messages []api.Message
+
+	// Add instructions as system message if present
+	if r.Instructions != "" {
+		messages = append(messages, api.Message{
+			Role:    "system",
+			Content: r.Instructions,
+		})
+	}
+
+	// Handle simple string input
+	if r.Input.Text != "" {
+		messages = append(messages, api.Message{
+			Role:    "user",
+			Content: r.Input.Text,
+		})
+	}
+
+	// Handle array of input items
+	// Track pending reasoning to merge with the next assistant message
+	var pendingThinking string
+
+	for _, item := range r.Input.Items {
+		switch v := item.(type) {
+		case ResponsesReasoningInput:
+			// Store thinking to merge with the next assistant message
+			pendingThinking = v.EncryptedContent
+		case ResponsesInputMessage:
+			msg, err := convertInputMessage(v)
+			if err != nil {
+				return nil, err
+			}
+			// If this is an assistant message, attach pending thinking
+			if msg.Role == "assistant" && pendingThinking != "" {
+				msg.Thinking = pendingThinking
+				pendingThinking = ""
+			}
+			messages = append(messages, msg)
+		case ResponsesFunctionCall:
+			// Convert function call to assistant message with tool calls
+			var args api.ToolCallFunctionArguments
+			if v.Arguments != "" {
+				if err := json.Unmarshal([]byte(v.Arguments), &args); err != nil {
+					return nil, fmt.Errorf("failed to parse function call arguments: %w", err)
+				}
+			}
+			msg := api.Message{
+				Role: "assistant",
+				ToolCalls: []api.ToolCall{{
+					ID: v.CallID,
+					Function: api.ToolCallFunction{
+						Name:      v.Name,
+						Arguments: args,
+					},
+				}},
+			}
+			// Attach pending thinking
+			if pendingThinking != "" {
+				msg.Thinking = pendingThinking
+				pendingThinking = ""
+			}
+			messages = append(messages, msg)
+		case ResponsesFunctionCallOutput:
+			messages = append(messages, api.Message{
+				Role:       "tool",
+				Content:    v.Output,
+				ToolCallID: v.CallID,
+			})
+		}
+	}
+
+	// If there's trailing reasoning without a following message, emit it
+	if pendingThinking != "" {
+		messages = append(messages, api.Message{
+			Role:     "assistant",
+			Thinking: pendingThinking,
+		})
+	}
+
+	options := make(map[string]any)
+
+	if r.Temperature != nil {
+		options["temperature"] = *r.Temperature
+	} else {
+		options["temperature"] = 1.0
+	}
+
+	if r.TopP != nil {
+		options["top_p"] = *r.TopP
+	} else {
+		options["top_p"] = 1.0
```

Comment:
I don't think we should set these sampling params here. These can be coming from a modelfile, and IIUC this would overwrite it.